server/api/appointment_handler.py

```diff
@@ -0,0 +1,130 @@
+import json, re
+from flask import jsonify, request, Blueprint, session
+from datetime import datetime
+from api.login_decorator import login_required
+from pytz import all_timezones
+
+# Models
+from models.shared import db
+from models.user import User
+from models.meeting import Meeting
+from models.appointment import Appointment
+
+EMAIL_REGEX = re.compile(r"[^@]+@[^@]+\.[^@]+")
+
+appointment_handler = Blueprint('appointment_handler', __name__)
+
+@appointment_handler.route('/appointments', methods=["GET"])
+@login_required
+def appointment_get():
+    sess = dict(session)
+    user = User.query.filter_by(email=sess["profile"]["email"]).first()
+
+
+    if user is None:
+        return jsonify({
+            "status": "error",
+            "message": "Please complete setting up your profile first."
+        }), 400
+
+    # We need to output the following:
+    # - name, description, duration
+    # Should we only display upcoming appointments or show everything?
```

Comment:
i think ideally we want to show both upcoming and previous - on the frontend, we mark upcoming/previous

Reply:
Will do, this should probably be limited as well in terms of the output, or no?

Reply:
yep limit this as well :)